sweepbatcher/sweep_batch.go

```diff
@@ -1318,11 +1336,50 @@ func constructUnsignedTx(sweeps []sweep, address btcutil.Address,
 	fee := clampBatchFee(feeForWeight, batchAmt)
 
 	// Add the batch transaction output, which excludes the fees paid to
-	// miners.
-	batchTx.AddTxOut(&wire.TxOut{
-		PkScript: batchPkScript,
-		Value:    int64(batchAmt - fee),
-	})
+	// miners. Reduce the amount by the sum of change outputs, if any.
+	if len(changeOutputs) == 0 {
```

Comment:
We don't need this special case, IMHO.
If there is no change, sumChange is 0 and the `else` branch works well.

Reply:
yes this is simpler, fixed